File: Data Structures & Algorithms/is-palindrome/submission-6.js
class Solution {

    isAlphaNumeric(c){
        return (
            (c >= "a" && c <= "z") ||
            (c >= "A" && c <= "Z") ||
            (c >= "0" && c <= "9")
        );
    }
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        let l = 0;
        let r = s.length - 1;

        while(l < r){
            while(l < r && !this.isAlphaNumeric(s[l])){
                l++;
            }

            while(r > l && !this.isAlphaNumeric(s[r])){
                r--;
            }

            if(s[l].toLowerCase() !== s[r].toLowerCase()){
                return false;
            }

            l++;
            r--;
        }

        return true;
    }
}

File: Data Structures & Algorithms/length-of-last-word/submission-6.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        length = 0
        r = len(s) - 1

        while (s[r] == " "):
            r -= 1
        while r >= 0 and s[r] != " ":
            r-=1
            length += 1

        return length
        